drivers/gpio/gpio_stm32.c

```diff
@@ -474,6 +474,12 @@ static int gpio_stm32_config(const struct device *dev,
 		goto exit;
 	}
 
+
+#ifdef CONFIG_PM_DEVICE_IDLE
```

Comment:
I wonder if it would be better to just add *dummy* PM functions when device PM is not enabled, so that we do not pollute drivers with tons of `#ifdef CONFIG_PM_XXX`. See, e.g. https://elixir.bootlin.com/linux/latest/source/include/linux/pm_runtime.h#L231 Opinions @ceolin @nashif ?

Reply:
I'm getting rid of CONFIG_PM_DEVICE_IDLE option, that should simplify all this `#ifdef` code. But I tend to agree with you about all these conditional code, we should avoid as much as possible. Dummy functions have to be carefully thought though, they will return an error code and these error codes have to be properly handled by the caller.